spyrmsd/rmsd.py

```diff
@@ -373,3 +373,181 @@ def rmsdwrapper(
             )
 
     return RMSDlist
+
+
+import os
+from concurrent.futures import ProcessPoolExecutor
+from functools import partial
+from multiprocessing import Process, Queue
+
+
+def _rmsd_queue(
+    molref: molecule.Molecule,
+    mols: Union[molecule.Molecule, List[molecule.Molecule]],
+    queue: Queue,
+    symmetry: bool = True,
+    center: bool = False,
+    minimize: bool = False,
+    strip: bool = True,
+    cache: bool = True,
+) -> Any:
+    """
+    Compute RMSD between two molecules and put it in a queue.
+
+    Parameters
+    ----------
+    molref: molecule.Molecule
+        Reference molecule
+    mols: Union[molecule.Molecule, List[molecule.Molecule]]
+        Molecules to compare to reference molecule
+    queue: Queue
+        The queue where to put the result
+    symmetry: bool, optional
+        Symmetry-corrected RMSD (using graph isomorphism)
+    center: bool, optional
+        Center molecules at origin
+    minimize: bool, optional
+        Minimised RMSD (using the quaternion polynomial method)
+    strip: bool, optional
+        Strip hydrogen atoms
+
+    Returns
+    -------
+    None
+        The RMSD result is put in the queue.
+    """
+    queue.put(
+        rmsdwrapper(
+            molref=molref,
+            mols=mols,
+            symmetry=symmetry,
+            center=center,
+            minimize=minimize,
+            strip=strip,
+            cache=cache,
+        )
+    )
+
+
+def rmsd_timeout(
+    molref: molecule.Molecule,
+    mols: Union[molecule.Molecule, List[molecule.Molecule]],
+    symmetry: bool = True,
+    center: bool = False,
+    minimize: bool = False,
+    strip: bool = True,
+    cache: bool = True,
+    timeout: Optional[float] = 5,  ## Do you think this is a good default timeout value?
```

Comment:
```suggestion
    timeout: Optional[float] = 5,
```
Good question! What about not setting a timeout by default (i.e. unlimited timeout)?

Reply:
Sure, at first I had it like that, but at that time I was thinking to keep this function public. With that in mind it seemed a bit counter-intuitive to me set the default timeout to None if the whole function's purpose is to have a limited timeout. However since we're making it private it doesn't really matter anyway!

Reply:
Oh, I see. Thanks for explaining. As I mentioned in another comment, I think that timeout and parallelism should be compacted in a single function (and I see that in the parallel function you indeed have an unlimited timeout).